src/ui/Menu/Menu.tsx

```diff
@@ -29,26 +33,68 @@ type MenuProps = RadixDropdownMenu.DropdownMenuProps & {
  * ```
  */
 const Menu: FC<MenuProps> = ({
-  button,
-  children,
+  trigger,
   menuPopupProps,
+  children,
   ...props
 }) => {
+  const [showMenu, setShowMenu] = useState(props.defaultOpen);
   return (
-    <RadixDropdownMenu.Root {...props}>
-      <RadixDropdownMenu.Trigger asChild>{button}</RadixDropdownMenu.Trigger>
-      <MenuContent align="start" sideOffset={8} {...menuPopupProps}>
-        {children}
-      </MenuContent>
+    <RadixDropdownMenu.Root
+      {...props}
+      open={showMenu}
+      onOpenChange={setShowMenu}
+    >
+      <RadixDropdownMenu.Trigger asChild>{trigger}</RadixDropdownMenu.Trigger>
+      <AnimatePresence>
+        {/* Required to aniamte exit */}
+        {showMenu && (
+          <MenuContent
+            align="start"
+            sideOffset={8}
+            {...menuPopupProps}
+            asChild
+            forceMount
+            portalled={false}
+          >
+            <motion.div
+              transition={ANIMATION_TRANSITION}
+              variants={ANIMATION_VARIANTS}
+              initial={props.defaultOpen ? false : `hidden`}
+              animate="visible"
+              exit="hidden"
+            >
+              {children}
+            </motion.div>
+          </MenuContent>
+        )}
+      </AnimatePresence>
     </RadixDropdownMenu.Root>
   );
 };
 
 export default Menu;
 
+/**
+ * Animation for expanding and collapsing effect of the menu
+ */
+const ANIMATION_TRANSITION: Transition = {
+  type: 'spring',
+  duration: 0.5,
+};
+const ANIMATION_VARIANTS: Variants = {
+  visible: {
+    height: 'auto',
+  },
+  hidden: {
+    height: 0,
+  },
+};
```

Comment:
this is a cool animation. Don't abuse animating height though, it's [costly](https://web.dev/animations-guide/#triggers).

Reply:
Yea, I understand that `height` is a CPU-calculated parameter while `transform` is GPU-calculated so it's more heavy. But what're the alternative here for this animation?

Reply:
I guess just don't use it on animation that may happen very frequently?

Reply:
To make the same animation? Not possible. Just don't use these kinds of animation too much, especially on mobile.